File: core/privacy/node/anonymizer.js
/**
 * PEAC Protocol Privacy Anonymizer (Node)
 * Apache 2.0 License
 */
const crypto = require('crypto');

/**
 * Hashes input (user/agent ID) with SHA-256.
 * @param {string} id - The input string (e.g., agent ID)
 * @returns {string} - Hex-encoded SHA-256 hash
 */
function anonymizeId(id) {
  return crypto.createHash('sha256').update(id).digest('hex');
}

/**
 * Logs a PEAC request, respecting the do_not_log flag.
 * @param {object} req - { agentId, path, do_not_log }
 * @returns {object|null} - Logged entry or null if not logged
 */
function logRequest({ agentId, path, do_not_log }) {
  if (do_not_log) return null;
  return {
    timestamp: new Date().toISOString(),
    agent: anonymizeId(agentId),
    path,
    privacy: do_not_log ? 'private' : 'normal',
  };
}

module.exports = { anonymizeId, logRequest };

File: archived/core/privacy/python/anonymizer.py
"""
PEAC Protocol Privacy Anonymizer (Python)
Apache 2.0 License
"""
import hashlib
from datetime import datetime

def anonymize_id(agent_id: str) -> str:
    """SHA-256 hash of agent_id as hex."""
    return hashlib.sha256(agent_id.encode()).hexdigest()

def log_request(agent_id, path, do_not_log=False):
    """Log a request unless do_not_log is set."""
    if do_not_log:
        return None
    return {
        "timestamp": datetime.utcnow().isoformat(),
        "agent": anonymize_id(agent_id),
        "path": path,
        "privacy": "private" if do_not_log else "normal"
    }
